### PR TITLE
Crud operations

### LINKED ISSUE #25: Implement CRUD Operations for Movie Entity
In this issue, we need to create a repository class to manage CRUD operations (Create, Read, Update, Delete) for the Movie entity in our application. The repository class should utilize Jakarta Persistence to interact with the database and enable easy access and management of movies in the database table.

### PR DESCRIPTION
**Pull request by TrungMa1**
I saw that Martin left this comment "Don't use primary key as id. Use nanoid or UUID" in the PersonResource file, so i changed our id to UUID instead. 

I also tried to follow the template of PersonRepository with our MovieRepository class. But im nowhere confident in my work so would appreciate some feedback. If its all wrong we can just erase the branch.

I did also move classes into the right packages.

this PR closes #25

### CODE REVIEW

**Review by Lucia560 — APPROVED** [review]

File: src/main/java/org/example/jakarta_labb/resource/MovieResource.java
Comment (marcusju23) [inline src/main/java/org/example/jakarta_labb/resource/MovieResource.java:17]: Might be good to change the @Inject annotation to a constructor injection instead. Sonar mentioned this as well as [Issue](https://sonarcloud.io/project/issues?resolved=false&pullRequest=26&id=Lucia560_Jakarta_labb&open=AY3dLnPWY7d5sJwlFNgd&tab=code). 

> By using constructor injection, the dependencies are explicit and must be passed during an object’s construction. This avoids the possibility of instantiating an object in an invalid state and makes types more testable. Fields can be declared final, which makes the code easier to understand, as dependencies don’t change after instantiation.

Read more [here](https://sonarcloud.io/project/issues?resolved=false&pullRequest=26&id=Lucia560_Jakarta_labb&open=AY3dLnPWY7d5sJwlFNgd&tab=how_to_fix) on how to fix this issue.

**Review by marcusju23 — CHANGES_REQUESTED** [review]
Great contribution! I came across one thing that might need adjustment before approval. What are your thoughts on it? 🤔

File: src/main/java/org/example/jakarta_labb/resource/MovieResource.java
Comment (TrungMa1) [reply src/main/java/org/example/jakarta_labb/resource/MovieResource.java:17]: i thought so too, but wanted to wait for confirmation! Will get right on it!

File: src/main/java/org/example/jakarta_labb/entity/Movie.java
Comment (BarbieQkiller) [inline src/main/java/org/example/jakarta_labb/entity/Movie.java:15]: I am not sure but I don't think GenerationType.IDENTITY works for UUID.
We are still exposing the primary key. Should we keep the long id and add a new field for UUID?

File: src/main/java/org/example/jakarta_labb/resource/MovieResource.java
Comment (BarbieQkiller) [inline src/main/java/org/example/jakarta_labb/resource/MovieResource.java:25]: I don't think we should expose the entity.
Please use the movie DTO in all endpoints 😃

**Review by BarbieQkiller — COMMENTED** [review]
Wow! Great job. My questions are coming from me being unsure and should not be read as critique but me trying to get a better grasp of things. Would be lovely to have all other members of the group chime in 😺 

Se comments for context.

**Comment by BarbieQkiller** [comment]
Heads up when encountering database, we opted for postgres and I noticed that mysql are referenced in the code.

File: src/main/java/org/example/jakarta_labb/entity/Movie.java
Comment (TrungMa1) [reply src/main/java/org/example/jakarta_labb/entity/Movie.java:15]: You're right! Maybe i can use "GenerationType.UUID" instead. Or as you mentioned, create a seperate field for UUID.

File: src/main/java/org/example/jakarta_labb/resource/MovieResource.java
Comment (TrungMa1) [reply src/main/java/org/example/jakarta_labb/resource/MovieResource.java:25]: was thinking i could take care of that in my next PR when we have a MovieDto class to refer to.

**Comment by TrungMa1** [comment]
> Heads up when encountering database, we opted for postgres and I noticed that mysql are referenced in the code.

would changing "mysql" to "postgresql" solve this?
@PersistenceContext(unitName = "postgresql")

File: src/main/java/org/example/jakarta_labb/resource/MovieResource.java
Comment (BarbieQkiller) [reply src/main/java/org/example/jakarta_labb/resource/MovieResource.java:25]: Sounds great!

File: src/main/java/org/example/jakarta_labb/entity/Movie.java
Comment (BarbieQkiller) [reply src/main/java/org/example/jakarta_labb/entity/Movie.java:15]: Sounds great to add a new column for UUID :)

**Comment by Lucia560** [comment]
A check has failed.  Is a action/v3 instead of v4 in sonar.yml. I will fix it in about an hour .

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Review by Lucia560 — DISMISSED** [review]
Jättebra jobbat, verkligen :)

**Review by Lucia560 — APPROVED** [review]

**Review by marcusju23 — APPROVED** [review]
Good job! 👍

**Pull request merged**
